functions/go/Makefile

```diff
@@ -0,0 +1,33 @@
+GOPATH := $(shell go env GOPATH)
+
+FUNCTIONS := \
```

Comment:
I imagine we should have a docker build target for building image: `gcr.io/kpt-functions-trusted/`+fn_name

When there are fn from external contributors, we may need to have a `curated_function` list and a `contrib_function`list. The contrib fn will go to `gcr.io/kpt-functions-contrib/`

Reply:
Currently functions in this catalog are named in this way: `gcr.io/kpt-functions/`+fn name.

@frankfarzan How do you think about this proposal.

Reply:
We're discussing this in @mengqiy's document, I'm advocating for a delineation of community contributed functions. I'll defer to you whether to make this change in a future PR though

Reply:
https://github.com/GoogleContainerTools/kpt/issues/1246